### PR TITLE
Add underscore prefix to all private members

### LINKED ISSUE #256: All "private" classes and members should have "_" prefix
These classes should not be used by clients, therefore we should prefix with and underscore. Examples are the attribute classes, etc

### PR DESCRIPTION
**Pull request by marcoskirsch**
[X] This contribution adheres to [CONTRIBUTING.md]

### What does this Pull Request accomplish?

* Fix #256 
* Adds underscore prefix to private members for consistency with PEP8 coding conventions
* Shrinks the "surface area" of API by signaling clients not to use those.
* Renames 'session_name' to 'session_handle_parameter_name' in ParamListType, consistent with config.py metadata.
* Renames 'name_to_use': 'library_call_name' in ParamListType to 'library_call_snippet', as it contains code not just a name.
* Fix a handful of locations where "vi" was still hardcoded, rather than use ${config['session_handle_parameter_name']}

### Why should this Pull Request be merged?

* Bug fixes
* Naming improvements

### What testing has been done?

tox

### CODE REVIEW

File: build/templates/session.py.mako [hunk 1]
```diff
@@ -158,8 +157,8 @@ class Session(_SessionBase):
     def __init__(${init_method_params}):
         super(Session, self).__init__(repeated_capability='')
         # TODO(marcoskirsch): private members should start with _
```
Comment (texasaggie97) [inline build/templates/session.py.mako]: You can remove this now

File: src/nimodinst/templates/session.py.mako [hunk 2]
```diff
@@ -143,9 +143,9 @@ class Session(object):
 
     def close(self):
         # TODO(marcoskirsch): Should we raise an exception on double close? Look at what File does.
-        if(self.handle != 0):
-            self._close_installed_devices_session(self.handle)
-            self.${config['session_handle_parameter_name']} = 0
+        if(self._handle != 0):
+            self._close_installed_devices_session(self._handle)
```
Comment (texasaggie97) [inline src/nimodinst/templates/session.py.mako]: These two lines should use `${config['session_handle_parameter_name']}`

File: src/nimodinst/templates/session.py.mako [hunk 3]
```diff
@@ -27,7 +27,7 @@ class AttributeViInt32(object):
         self._attribute_id = attribute_id
 
     def __getitem__(self, index):
-        return self._owner._get_installed_device_attribute_vi_int32(self._owner.handle, self._index, self._attribute_id)
+        return self._owner._get_installed_device_attribute_vi_int32(self._owner._handle, self._index, self._attribute_id)
```
Comment (texasaggie97) [inline src/nimodinst/templates/session.py.mako]: Use `${config['session_handle_parameter_name']}`

File: src/nimodinst/templates/session.py.mako [hunk 4]
```diff
@@ -38,7 +38,7 @@ class AttributeViString(object):
         self._attribute_id = attribute_id
 
     def __getitem__(self, index):
-        return self._owner._get_installed_device_attribute_vi_string(self._owner.handle, self._index, self._attribute_id)
+        return self._owner._get_installed_device_attribute_vi_string(self._owner._handle, self._index, self._attribute_id)
```
Comment (texasaggie97) [inline src/nimodinst/templates/session.py.mako]: Use `${config['session_handle_parameter_name']}`

File: build/templates/session.py.mako (hunk 1 above)
Comment (marcoskirsch) [reply build/templates/session.py.mako]: fixed

File: src/nimodinst/templates/session.py.mako (hunk 3 above)
Comment (marcoskirsch) [reply src/nimodinst/templates/session.py.mako]: done

File: src/nimodinst/templates/session.py.mako (hunk 4 above)
Comment (marcoskirsch) [reply src/nimodinst/templates/session.py.mako]: fixed

File: src/nimodinst/templates/session.py.mako (hunk 2 above)
Comment (marcoskirsch) [reply src/nimodinst/templates/session.py.mako]: fixed

**Pull request merged**